packages/graphql/tests/integration/issues/464.int.test.ts

```diff
@@ -0,0 +1,172 @@
+            ${typeBook.operations.create} (
+                input: [
+                    {
+                        id: $id
+                        name: $name
+                        author: {
+                            create: {
+                                node: {
+                                    id: $authorId
+                                    name: $authorName
+                                }
+                            }
+                        }
+                    }
+                ]
+            ) {
+                info {
+                    bookmark
+                }
+                ${typeBook.plural} {
+                    id
+                    name
+                }
+            }
+        }
+    `;
+
+    const queryBooks = `
+        query {
+            ${typeBook.plural} {
+                id
+                name
+            }
+        }
+    `;
+
+    beforeAll(async () => {
+        driver = await neo4j();
+    });
+
+    afterAll(async () => {
+        await driver.close();
+    });
+
+    test("should run the mutation but not commit until it is done explicitly", async () => {
+        const session = driver.session();
+        const transaction = session.beginTransaction();
+        try {
+            const result = await graphql({
+                schema: await neoSchema.getSchema(),
+                source: createMutation,
+                contextValue: { driver, executionContext: transaction },
+                variableValues: {
+                    id: bookId,
+                    name: bookName,
+                    authorId,
+                    authorName,
+                },
+            });
+
+            if (result.errors) {
+                console.log(JSON.stringify(result.errors, null, 2));
+            }
+
+            expect(result.errors).toBeFalsy();
+
+            expect(result.data).toEqual({
+                [typeBook.operations.create]: {
+                    info: { bookmark: null },
+                    [typeBook.plural]: [
+                        {
+                            id: bookId,
+                            name: bookName,
+                        },
+                    ],
+                },
+            });
+
+            expect(transaction.isOpen()).toBe(true);
+
+            const noBooks = await graphql({
+                schema: await neoSchema.getSchema(),
+                source: queryBooks,
+                contextValue: { driver },
+            });
+
+            expect(noBooks.data?.[typeBook.plural]).toEqual([]);
+
+            await transaction.commit();
+
+            const books = await graphql({
+                schema: await neoSchema.getSchema(),
+                source: queryBooks,
+                contextValue: { driver },
+            });
+
+            expect(books.data?.[typeBook.plural]).toEqual([{ id: bookId, name: bookName }]);
+        } finally {
+            await session.close();
+        }
+    });
+
+    // @TODO: more tests
```

Comment:
👀  Do we need more tests here? I note there's no integration tests for using either a driver or a session in the `executionContext` field.

Reply:
More tests are always good I guess. As you point out there should be tests for using driver and session, but for our team only transaction is necessary. That is also the issue, that this PR is targeted at.

Reply:
Totally get that,  but if a PR is introducing new functionality then it all needs to be tested. Hopefully you can add them, but do give us a shout if you don't have time and we can get them added in.